modules/core/src/cuda_stream.cpp

```diff
@@ -312,6 +313,14 @@ cv::cuda::Stream::Impl::Impl(const Ptr<GpuMat::Allocator>& allocator) : stream(0
     ownStream = true;
 }
 
+cv::cuda::Stream::Impl::Impl(const int cudaFlags) : stream(0), ownStream(false)
+{
+    CV_DbgAssert(cudaFlags >= 0);
```

Comment:
IMHO, this check doesn't help.

My suggestion is to remove that.

Reply:
I do believe we need a limit test somewhere. It is illegal to pass negative numbers. Since the python shim infrastructure has a bug/issue (gives non-important warnings), I can't use unsigned int. This was discussed multiple times before.